src/app/[locale]/client-page.tsx

```diff
@@ -0,0 +1,39 @@
+'use client';
+
+import {useQuery} from '@tanstack/react-query';
+import {SearchResult} from '@/lib/dataset-fetcher';
+import DatasetCard from '@/components/dataset-card';
+import {Dataset} from '@/lib/dataset-fetcher';
+
+interface ClientPageProps {
+  initialDatasets: SearchResult;
```

Comment:
`initialDatasets` could suggest that it contains datasets, but it actually contains a search result, not just datasets. Change the name?